src/tools/kata-ctl/src/check.rs

```diff
@@ -89,13 +95,49 @@ pub fn run_network_checks() -> Result<()> {
     Ok(())
 }
 
+fn get_kata_version_by_url(url: &str) -> Result<String, reqwest::Error> {
+    let content = reqwest::blocking::Client::new()
+        .get(url)
+        .header(CONTENT_TYPE, "application/json")
+        .header(USER_AGENT, "kata")
+        .send()?
+        .json::<HashMap<String, Value>>()?;
+
+    let version = content["tag_name"].as_str().unwrap();
```

Comment:
This is valid, but it's best to avoid `unwrap()`. However, removing this one is going to be awkward, so I guess it'll have to stay :smile:

Reply:
:)

Reply:
@jodh-intel Question, will this not panic in case there is an error? I dont see a path where we are returning reqwest::Error.

Reply:
@amshinde - it will, and I'd rather we don't have any `unwrap()` calls at all tbh. However, remove that one is awkward because this function returns a `reqwest::Error` and there isn't an easy way to construct a generic one fwics. We could change this function to return an `anyhow::Result<()>`, but that breaks the error checking logic for the handler.

I'd really like to get this PR landed so we can raise a PR on main using #5334 - we already have folk waiting to contribute fwics... ;) So unless we can find a quick solution to this, I'm happy to leave it and we can ponder the problem and interate on the code in `main`.